### ISSUE TITLE
(de)serialize Double <-> String

### ISSUE THREAD

**Issue by nimo23** [eclipse-ee4j/yasson#306]
I get this error when trying to (de)serialize the primitive type `double`:

```
13:11:18,793 SEVERE [org.eclipse.yasson.internal.Marshaller] double value cannot be Infinite or NaN
13:11:18,803 SEVERE [org.eclipse.yasson.internal.Marshaller] Generating incomplete JSON
13:11:18,803 ERROR [io.util.EntityTool] javax.json.bind.JsonbException: Internal error: double value cannot be Infinite or NaN
```

When I change `double` to `Double` all works. 
Am I forced to change all the primitive "`double`" properties to "`Double`" only to work with yasson?

_Btw, the error log does not show **which** property makes trouble..but it's a property with the type "double"_

**Comment by aguibert** [comment]
hi @nimo23, this issue seems to be an exact duplicate of this other issue you raised earlier:
https://github.com/eclipse-ee4j/yasson/issues/297

**Comment by aguibert** [comment]
> Am I forced to change all the primitive "double" properties to "Double" only to work with yasson?

No, either one is acceptable. The issue is (as the error message indicates) you can't serialize/deserialize infinite/NaN values into double/Double

**Comment by nimo23** [comment]
>this issue seems to be an exact duplicate of this other issue you raised earlier:

No duplicate, the other was about "show the property  which makes trouble"..
this issue is about "double/Double conversion"..

**You say:** 

> you can't serialize/deserialize infinite/NaN values into double/Double

Is this okay? Why not (de)serializing a NaN value to NaN?

`Double test = Double.NaN;`

NaN should be converted to what it is: a `NaN`:
```
{
..
test: "NaN"
..
}


```

**Comment by nimo23** [comment]
I actually looked at what Jackson do with such values. Jackson does what it should!
```

{
  "val1" : "NaN",
  "val2" : 1.0,
  "val3" : 0.0,
  "val4" : "Infinity"
}
```

Please re-open this issue.

**Comment by aguibert** [comment]
Jackson produces values for infinity and Nan, but these are represented as Strings, not numbers, which would likely break clients that are expecting a number value here.

The JSON-B spec builds on top of the JSON-P spec, and currently the JSON-P spec mandates that these values throw an exception:
```java
    /**
     * Writes the specified value as a JSON number value within the current
     * array, field or root context. The string {@code BigDecimal.valueOf(value).toString()}
     * is used as the text value for writing.
     *
     * @param value a value to be written in current JSON array
     * @return this generator
     * @throws javax.json.JsonException if an i/o error occurs (IOException
     * would be cause of JsonException)
     * @throws JsonGenerationException if this method is not called within an
     *      array or root context.
     * @throws NumberFormatException if the value is Not-a-Number (NaN) or infinity.
     */
    JsonGenerator write(double value);
```

There is also a more general problem for double values that are outside of the range outside of Java's `double` primitive. This issue is being discussed at the JSON-P level here:
https://github.com/eclipse-ee4j/jsonp/issues/160

So while I think this is a valid topic that needs to be discussed, the proper place to resolve this is at the JSON-P spec level in issue 160, rather than here in Yasson.

**Comment by nimo23** [comment]
Could Json-B provides a workaround as long as Json-P does not handle Double values  correctly? 

**Could you build a build-in `Converter` to provide `NaN` and `Infinty` for the Wrapper `Double` within Json-B? Json-B could use that Converter behind the (de)serializing process when detecting the wrapper type `Double`.** This would make sense, dont you think so?


Actually, I cannot use Yasson **because it does not handle Double right,** I am forced to switch back to Jackson because of this.

**Comment by m0mus** [comment]
I don't agree that it should be solved at JSONP level. I think we should consider it as an enhancement for the new JSONB spec or as a Yasson specific feature.

**Comment by tsfullman** [comment]
I am having this issue too and would love for the de-serialization of Infinity to work correctly similar to Jackson
